### ISSUE TITLE
How to update reference files on Windows

### ISSUE THREAD

**Issue by faze-geek** [lcompilers/lpython#1462]
This may seem like an issue that can be discussed on zulip. I have already done that through PM's, will close this as soon as I get required solution.

I am trying to follow this [comment](https://github.com/lcompilers/lpython/pull/1430#discussion_r1072285841) by @czgdp1807 .

I believe that when I am trying to update the reference files, all of the reference files get updated which is unexpected. I guess this is because I am not running the windows equivalent of `rm ./*` after `cd tests/reference`. Could you kindly help me figuring out a windows equivalent for that command.

I'll attach screen shots for clarity on what is going on.
Image_1 shows all reference files being updated instead of only the required ones.
Image_2 shows commands that I am running to update( where `rm ./*` is intended to delete unwanted files)

<img width="344" alt="image" src="https://user-images.githubusercontent.com/90216905/214474564-1f2b314c-e6aa-4e27-a9ad-a2df9d67778c.png">

<img width="268" alt="image" src="https://user-images.githubusercontent.com/90216905/214474702-a33d77a4-ef5b-41c3-8b95-39ea4cfe2ede.png">

**Comment by czgdp1807** [comment]
What you can do is commit the changes you have made so far. Update your `main` branch locally, from `origin`. Then merge `main` into your working branch. Do a clean build. And run `python run_tests.py -u`.

**Comment by Thirumalai-Shaktivel** [comment]
Currently, `run_tests.py -u` doesn't work as expected in Windows. i.e., Both Linux and macOS get updated with the same output but this isn't the case in Windows. This is the issue related to the `\n\r` or simply `\n` is not handled correctly in Windows (or possibly some other issue.). This was the reason for multiple files updates like this: [`9628b56` (#1430)](https://github.com/lcompilers/lpython/pull/1430/commits/9628b56f7f976e69c8622f737ab6bd6841db250b#diff-0890e3a864c0af63820e9e74cdb2fc992db3b29f4f5a22b125125d4ac512e58e)

Also, regarding the `rm ./*`, be very careful while using this command, it can remove all the files in the current directory.
I use this: `rm tests\references\*` command only during the `git rebase` (which caused conflicts in the references) or while modifying the options name in run_tests. The rest of the time, I only do `run_tests.py -u`.

**Comment by faze-geek** [comment]
I was under the assumption that #1463 was an attempt to fix `python run_test.py -u` but it doesn't and all the reference files get updated (I have fetched the latest changes). 
If I only run `python run_test.py` it does show me the changes the `stdout_hash` should have in the form of errors but then I have to make those manually which I would not prefer.

**Comment by certik** [comment]
#1463 fixed `run_tests.py`, but not `run_tests.py -u`. @faze-geek would you be interested in sending us a PR that fixes `run_tests.py -u` on windows? That would be a big help.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, https://github.com/lcompilers/lpython/pull/1463 fixed `run_tests.py`. But, I don't know why it didn't fix `run_tests.py -u`. Let me check and report back.

**Comment by Thirumalai-Shaktivel** [comment]
I see, the issue is with the `git clone`. So, previously while you cloned the lpython repo all the files were converted to CRLF format. That is the reason for the failure.

@faze-geek why don't you try this?
- Commit the changes and publish or push the changes of the branches that you were working on if any.
- Delete the lpython directory. Clone it again (make sure the repo contains the `.gitattributes` file in it). Build and test it with `python run_tests.py --skip-run-with-dbg`.

If you face any issue with the `python run_tests.py -u`, please report it here.

**Comment by faze-geek** [comment]
> If you face any issue with the `python run_tests.py -u`, please report it here.

Sure, I will try the modifications you suggested soon.  I will also test if `run_tests.py` is working correctly on my windows machine.

**Comment by faze-geek** [comment]
> I see, the issue is with the `git clone`. So, previously while you cloned the lpython repo all the files were converted to CRLF format. Even if you fetched the latest changes, it wouldn't update the file format. That might be the reason for the failure.

@Thirumalai-Shaktivel I tried this suggestion, this time the reference files get updatedaccordingly. I made a small test pr with minimal change so that I can test it out #1515 .

Once you have a look at it and give me a green flag, I'll close this issue. Thanks!

**Comment by faze-geek** [comment]
Closing as completed in #1463 and `Readme` updated by this [commit](https://github.com/lcompilers/lpython/pull/1581/commits/3d9323540c32a18d84aa1f175d10913549fa829d)
